protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -284,9 +289,16 @@ func (rpccs *RPCConsumerServer) SendRelay(
 	retries := uint64(0)
 	timeouts := 0
 	unwantedProviders := rpccs.GetInitialUnwantedProviders(directiveHeaders)
+
+	forceCacheRefresh := false
+	_, ok := directiveHeaders[common.EXTENSION_OVERRIDE_HEADER_NAME]
+	if ok {
+		forceCacheRefresh = true
+	}
+
 	for ; retries < MaxRelayRetries; retries++ {
 		// TODO: make this async between different providers
-		relayResult, err := rpccs.sendRelayToProvider(ctx, chainMessage, relayRequestData, dappID, consumerIp, &unwantedProviders, timeouts)
+		relayResult, err := rpccs.sendRelayToProvider(ctx, chainMessage, relayRequestData, dappID, consumerIp, &unwantedProviders, timeouts, forceCacheRefresh)
```

Comment:
let's avoid adding this as an argument, we want directive headers in the chainMessage object